api/logs/v1/labels_enforcer.go

```diff
@@ -178,3 +182,18 @@ func initAuthzMatchers(lm []*labels.Matcher) ([]*labels.Matcher, error) {
 
 	return lm, nil
 }
+
+// AllowedNamespaces returns the list of namespaces that the user is allowed to list.
+func AllowedNamespaces(ctx context.Context) []string {
+	matchers := ctx.Value(matchersContextKey{})
+	if matchers == nil {
+		return nil
+	}
+	matchersTyped := matchers.(AuthzResponseData)
+
+	var namespaces []string
+	for _, m := range matchersTyped.Matchers {
```

Comment:
Does `matchersTyped.Matchers` always contain a list of `|`-separated namespace names?
It's not directly related to this PR, but I wonder why the attribute is called `Matchers` and not `Namespaces`.

Reply:
It comes from the Prometheus API. I think this is what the OPA openshift returns

Reply:
https://github.com/observatorium/opa-openshift?tab=readme-ov-file#post-v1datapackagerule